admin/listPurchases/listView.html

```diff
@@ -83,28 +83,89 @@
+        if(startIndexToRender + Math.floor(amountOfShownSubpages/2) >= amountOfSubpages){
+            startIndexToRender = amountOfSubpages - amountOfShownSubpages;
+        }else if(startIndexToRender - Math.floor(amountOfShownSubpages/2) > 0){
+            startIndexToRender -= Math.floor(amountOfShownSubpages/2);
+        }else{
+            startIndexToRender = 0;
+        }
+    }
+    
+    for(let i = startIndexToRender; i < startIndexToRender + amountOfShownSubpages && i < amountOfSubpages; i++){
+        nav.innerHTML += `<a href="#" class="option subpage-selector subpage-selector-index-${i}">${i+1}</a>`
+    }
+
+    if(amountOfSubpages >= amountOfShownSubpages){
+        nav.innerHTML += `<a href="#" class="option subpage-selector subpage-selector-forward">></a>`
+        nav.innerHTML += `<a href="#" class="option subpage-selector subpage-selector-to-end">>></a>`
+        document.querySelector(`.subpage-selector-backwards`).addEventListener("click", function (e){
+            e.preventDefault();
+            if(entryPageIndex <= 0) return;
+            entryPageIndex--;
+            updateContent(option);
+            generateSubPageNavigation(option);
+        });
+        document.querySelector(`.subpage-selector-forward`).addEventListener("click", function (e){
+            e.preventDefault();
+            if(entryPageIndex >= amountOfSubpages - 1) return;
+            entryPageIndex++;
+            updateContent(option);
+            generateSubPageNavigation(option);
+        });
+        document.querySelector(`.subpage-selector-to-start`).addEventListener("click", function (e){
+            e.preventDefault();
+            entryPageIndex = 0;
+            updateContent(option);
+            generateSubPageNavigation(option);
+        });
+        document.querySelector(`.subpage-selector-to-end`).addEventListener("click", function (e){
+            e.preventDefault();
+            entryPageIndex = amountOfSubpages - 1;
+            updateContent(option);
+            generateSubPageNavigation(option);
+        });
+    }
+    
+    for(let i = startIndexToRender; i < startIndexToRender + amountOfShownSubpages && i < amountOfSubpages; i++){
+        document.querySelector(`.subpage-selector-index-${i}`).addEventListener("click", function (e){
+            e.preventDefault();
+            document.querySelector(`.subpage-selector-index-${entryPageIndex}`).classList.remove("selected");
+            entryPageIndex = i;
+            document.querySelector(`.subpage-selector-index-${entryPageIndex}`).classList.add("selected");
+            updateContent(option);
+            generateSubPageNavigation(option);
+        });
+        
+    }
+    
+    document.querySelector(`.subpage-selector-index-${entryPageIndex}`).classList.add("selected");
+}
+
+function updateContent(option){
+    renderEntriesToList(entries.filter(option.filter).slice(entryPageIndex * 10, entryPageIndex * 10 + 10));
```

Comment:
Split this into more lines for readability. This line is very hard to read because too much happens!